src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -413,54 +406,66 @@ private void handleMetricsTag(InputFile inputFile) throws XMLStreamException {
+      saveMetric(inputFile, metric, value);
+    }
+
     private void handleLinesMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.LINES, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.LINES);
     }
 
     private void handleClassesMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.CLASSES, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.CLASSES);
     }
 
     private void handleAccessorsMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.ACCESSORS, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.ACCESSORS);
     }
 
     private void handleStatementsMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.STATEMENTS, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.STATEMENTS);
     }
 
     private void handleFunctionsMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.FUNCTIONS, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.FUNCTIONS);
     }
 
     private void handlePublicApiMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.PUBLIC_API, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.PUBLIC_API);
     }
 
     private void handlePublicUndocumentedApiMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.PUBLIC_UNDOCUMENTED_API, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.PUBLIC_UNDOCUMENTED_API);
     }
 
     private void handleComplexityMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.COMPLEXITY, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.COMPLEXITY);
     }
 
     private void handleFileComplexityDistributionMetricTag(InputFile inputFile) throws XMLStreamException {
-      String value = stream.getElementText();
-      context.saveMeasure(inputFile, new Measure(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION, value).setPersistenceMode(PersistenceMode.MEMORY));
+      parseAndSaveStringMetric(inputFile, CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION);
+//      String value = stream.getElementText();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)